huaweicloud/services/acceptance/cfw/resource_huaweicloud_cfw_firewall_test.go

```diff
@@ -0,0 +1,322 @@
+package cfw
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func getFirewallResourceFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	region := acceptance.HW_REGION_NAME
+	// getFirewall: Query the List of CFW firewalls
+	var (
+		getFirewallHttpUrl = "v1/{project_id}/firewall/exist"
+		getFirewallProduct = "cfw"
+	)
+	getFirewallClient, err := cfg.NewServiceClient(getFirewallProduct, region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating CFW client: %s", err)
+	}
+
+	getFirewallPath := getFirewallClient.Endpoint + getFirewallHttpUrl
+	getFirewallPath = strings.ReplaceAll(getFirewallPath, "{project_id}", getFirewallClient.ProjectID)
+	getFirewallPath += fmt.Sprintf("?offset=0&limit=10&service_type=0&fw_instance_id=%s", state.Primary.ID)
+
+	getFirewallOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	getFirewallsResp, err := getFirewallClient.Request("GET", getFirewallPath, &getFirewallOpt)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving Firewalls: %s", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("error retrieving fire walls: %s", err)
```

Please check the other logging message.

Reply:
fixed